src/hooks/api/index.ts

```diff
@@ -28,13 +28,34 @@ export function useMainCurrency(): SWRResponse<Commodity> {
   );
 }
 
-export function useCommodities(): SWRResponse<Commodity[]> {
-  const key = '/api/commodities';
-  return useSWRImmutable(
+export function useCommodities(
+  guid?: string,
+): SWRResponse<Commodity[] | Commodity> {
+  let key = '/api/commodities';
+  if (guid) {
+    key = `${key}/${guid}`;
+  }
+
+  const result = useSWRImmutable(
     key,
-    // Needs to be encapsulated in arrow function or it doesnt work properly
-    fetcher(async () => Commodity.find(), key),
+    fetcher(() => Commodity.findBy({ guid }), key),
   );
+
+  if (result.error) {
+    throw new Error(result.error);
+  }
+
+  if (result.data && !guid) {
+    result.data.forEach(
+      (commodity: Commodity) => mutate(
+        `/api/commodities/${commodity.guid}`,
+        commodity,
+        { revalidate: false },
```

Comment:
this is necessary or otherwise we get an infinite loop when rendering a detail page for some cases where we have a page asking for the list and the detail keys (i.e AccountSelector which is loaded in the Topbar)